packages/headless/src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts

```diff
@@ -107,7 +107,9 @@ export const buildBreadcrumbManager = (
             dispatch(
               executeDeselectAllCategoryFacetValues({
                 facetId,
-                numberOfValues: 5,
+                numberOfValues:
+                  engine.state.categoryFacetSet[facetId].currentValues[0]
+                    .retrieveCount,
```

Comment:
Can't use `engine.state.categoryFacetSet[facetId]` because it is set to `1` in `toggleSelectCategoryFacetValue`. 
But the original `numberOfValues` is passed to the current values as the `retrieveCount`. And `currentValues[0]` is always defined, otherwise, there would be no breadcrumb to deselect.

Reply:
>  because it is set to 1 in toggleSelectCategoryFacetValue.

Can you give more context as to what this comment means exactly ? I'm not sure I understand.

Is it normal that it's set to 1 ? 🤔

Reply:
oops, forgot to add `.numberOfValues` 😅 I updated the comment above

Reply:
Setting the `numberOfValues` to 1 is needed by the api atm. Here is an example of the interaction:

Initial request (nothing selected)
```
currentValues = []
numberOfValues = 5
```
returns up to 5 values for the facet in the response

When one value is selected
```
currentValues = [selected]
numberOfValues = 1 // if not one, api returns an error.
```
The number of values for this request is determined by the `retrieveCount` property on the `selected` value.

Reply:
Because of this overwrite, we lose the initial number of values. We could fix this by refactoring the state to store the initial options separately, but this will be a large refactor.

I think using the retrieve count is a pragmatic work-around. If we internalize this into the slice reducer, someone dispatching the `executeDeselectAllCategoryFacetValues` action should not need to pass the number values. There is one scenario where the work-around would not work, when number of values is not equal to the retrieve count.


I think the best option would be to never change the `numberOfValues` in state, and rather set it dynamically when sending the request if we detect that current values is populated.

I believe this is the best path, because it would also allow us to internalize the `numberOfValues` logic for several actions in this PR https://github.com/coveo/ui-kit/pull/350. This is one less parameter, and will make the actions easier to use.

The draw-back is we are hard-coding a UX rule into headless, that the initial number of values and the subsequent "retrieve count" must always be the same. I personally think being opinionated and less flexible as a positive.